test/integration/imagestream_admission_test.go

```diff
@@ -309,64 +307,86 @@ func TestImageStreamAdmitSpecUpdate(t *testing.T) {
 	}
 
 	t.Logf("adding new tag to image stream spec exceeding limit %v", limit)
-	is, err := client.Image().ImageStreams(testutil.Namespace()).Get("is", metav1.GetOptions{})
+	is, err := client.ImageV1().ImageStreams(testutil.Namespace()).Get("is", metav1.GetOptions{})
 	if err != nil {
 		t.Fatal(err)
 	}
-	is.Spec.Tags["tag2"] = imageapi.TagReference{
+	upsertSpecTag(&is.Spec.Tags, imagev1.TagReference{
 		Name: "tag2",
-		From: &kapi.ObjectReference{
+		From: &corev1.ObjectReference{
 			Kind: "ImageStreamTag",
 			Name: "src:tag2",
 		},
-	}
-	_, err = client.Image().ImageStreams(testutil.Namespace()).Update(is)
+	})
+	_, err = client.ImageV1().ImageStreams(testutil.Namespace()).Update(is)
 	if err == nil {
 		t.Fatalf("unexpected non-error")
 	}
 	if !quotautil.IsErrorQuotaExceeded(err) {
 		t.Errorf("expected quota exceeded error, got instead: %v", err)
 	}
-	for _, res := range []kapi.ResourceName{imageapi.ResourceImageStreamTags, imageapi.ResourceImageStreamImages} {
+	for _, res := range []corev1.ResourceName{imagev1.ResourceImageStreamTags, imagev1.ResourceImageStreamImages} {
 		if !strings.Contains(err.Error(), string(res)) {
 			t.Errorf("expected resource %q in error string: %v", res, err)
 		}
 	}
 
 	t.Logf("re-tagging the image under different tag")
-	is, err = client.Image().ImageStreams(testutil.Namespace()).Get("is", metav1.GetOptions{})
+	is, err = client.ImageV1().ImageStreams(testutil.Namespace()).Get("is", metav1.GetOptions{})
 	if err != nil {
 		t.Fatal(err)
 	}
-	is.Spec.Tags["1again"] = imageapi.TagReference{
+	upsertSpecTag(&is.Spec.Tags, imagev1.TagReference{
 		Name: "1again",
-		From: &kapi.ObjectReference{
+		From: &corev1.ObjectReference{
 			Kind: "ImageStreamTag",
 			Name: "src:tag1",
 		},
-	}
-	_, err = client.Image().ImageStreams(testutil.Namespace()).Update(is)
+	})
+	_, err = client.ImageV1().ImageStreams(testutil.Namespace()).Update(is)
 	if err != nil {
 		t.Fatalf("unexpected error: %v", err)
 	}
 }
 
+func upsertSpecTag(tags *[]imagev1.TagReference, tagReference imagev1.TagReference) {
+	for i := range *tags {
+		curr := (*tags)[i]
+		if curr.Name == tagReference.Name {
+			(*tags)[i] = tagReference
+			return
+		}
+	}
+	*tags = append(*tags, tagReference)
+}
+
+func upsertStatus(tags *[]imagev1.NamedTagEventList, tagEventList imagev1.NamedTagEventList) {
```

Comment:
nit: be consistent with naming, since the other is `upsertSpecTag` this one should be `upsertStatusTag`.